### DISCUSSION TITLE
"quantisation failed"

### DISCUSSION THREAD

**Discussion by osfa** [libvips/libvips#3893] [General]
Hello!
Trying to convert a large 28800 × 64800px jpeg to 8 color palette  png and getting "quantisation failed". Works with a much smaller image so somehow related to image size? Any ideas on how to get a better error out of this? pretty sure I'm not ooming with what mem-usage.

```
vips pngsave "my-input.jpeg" "dither-test.png" --palette=TRUE --compression=9 --Q=0 --vips-leak

quantise: quantisation failed
vips_threadset_free: peak of 8 threads
memory: high-water mark 7,11 GB
error buffer: quantise: quantisation failed

```

**Environment**

 - OS: MacOS 10.13, M2
 - vips-8.15.1

**Comment by jcupitt** [comment]
Heuy @osfa, I've never tried it on an image that large, you're unlikely to get a good result, I'd think. 

I tried a few sizes here -- 20,000 x 20,000 works (and needs 2gb), 20,000 x 30,000 fails. Maybe you're hitting an int overflow inside the palette optimiser?

It would be better to downsize to something small (1000 x 1000?), quantise that, then extract the palette and use that to dither the full-res image. Sadly this isn't supported by libvips right now.

**Reply by osfa** [reply]
Thanks for the fast response! I'll find a way around it, thanks again!
